src/client/app/effects/github.ts

```diff
@@ -143,11 +143,17 @@ export class GitHubEffects {
 
                 return { gist: gist, snippetId: snippetId };
             })
-            .mergeMap(({ gist, snippetId }) => Observable.from([
+            .mergeMap(({ gist, snippetId }) => {
+                // Add snippet as trusted if it isn't already
+                if (!trustedSnippetManager.isSnippetTrusted(snippetId, gist.id)) {
```

Comment:
Realistically we're now creating a small security loophole -- that if you import someone's snippet, then share it, it's suddenly trusted, even if you never explicitly trusted the snippet in the first place.

I would make this conditional based on whether the snippet, before sharing, was one of the following:
* Was already trusted (in which case, no-op)
* Had a gist field that was empty
* Was a snippet that belongs to the user already (and hence is implicitly trusted)

Reply:
Added.